### PR TITLE
🌱 Improve logging in test/framework

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

* Adjusts the framework logger to append to files instead of truncating its previous content
* `clusterctl move` via test-framework now also prints the command (similar to `clusterctl init`)
* `clusterctl move` now logs to the `logs/[cluster-namespace]/clusterctl-move.log` directory instead of `clusterctl-move.log`. This allows to distinct the log files for a specific test. (Previously only the last `clusterctl move` output was available)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
nit: GitHub markdown just drops `<>` in the PR description without escaping

File: test/framework/clusterctl/logger/log_file.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func CreateLogFile(input CreateLogFileInput) *LogFile {
 	filePath := filepath.Join(input.LogFolder, input.Name)
 	Expect(os.MkdirAll(filepath.Dir(filePath), 0750)).To(Succeed(), "Failed to create log folder %s", filepath.Dir(filePath))
 
-	f, err := os.Create(filePath) //nolint:gosec // No security issue: filepath is safe.
+	f, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666) //nolint:gosec // No security issue: filepath is safe.
```
Comment (sbueringer) [inline test/framework/clusterctl/logger/log_file.go:39]: We should probably change the func name as well. WDYT about `OpenLogFile`?

File: test/framework/clusterctl/client.go [hunk 2]
```diff
@@ -315,11 +316,17 @@ func Move(ctx context.Context, input MoveInput) {
 	Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. input.ClusterctlConfigPath must be an existing file when calling Move")
 	Expect(input.FromKubeconfigPath).To(BeAnExistingFile(), "Invalid argument. input.FromKubeconfigPath must be an existing file when calling Move")
 	Expect(input.ToKubeconfigPath).To(BeAnExistingFile(), "Invalid argument. input.ToKubeconfigPath must be an existing file when calling Move")
-	Expect(os.MkdirAll(input.LogFolder, 0750)).To(Succeed(), "Invalid argument. input.LogFolder can't be created for Move")
+	logDir := path.Join(input.LogFolder, "logs", input.Namespace)
+	Expect(os.MkdirAll(logDir, 0750)).To(Succeed(), "Invalid argument. input.LogFolder can't be created for Move")
 
 	By("Moving workload clusters")
+	log.Logf("clusterctl --from-kubeconfig %s --to-kubeconfig %s --namespace %s",
```
Comment (fabriziopandini) [inline test/framework/clusterctl/client.go:323]: nit
```suggestion
	log.Logf("clusterctl move --from-kubeconfig %s --to-kubeconfig %s --namespace %s",
```

**Review by fabriziopandini — COMMENTED** [review]
overall lgtm pending nits
thanks for improving our test framework!

File: test/framework/clusterctl/client.go (hunk 2 above)
Comment (chrischdi) [reply test/framework/clusterctl/client.go:323]: Great find 🤦

File: test/framework/clusterctl/logger/log_file.go (hunk 1 above)
Comment (chrischdi) [reply test/framework/clusterctl/logger/log_file.go:39]: Should we create a new func then too?

**Comment by sbueringer** [comment]
/approve
(given Fabrizio's finding is fixed and I want to verify if merging still works after the lint test-infra config change)

**Pull request merged**